README.md
---
description: "Not Idea UI is an Open Source CSS framework based and created with TailwindCSS, with components created to use everywhere, easily and simply."
---

# Installation

You have different options to install it with **npm or yarn**.

## Getting Started

### As dependency

If you want to use it with TailwindCSS adding your own styles and color palettes.

You have to install it with `npm install not-idea-ui` or `yarn add not-idea-ui`. After, you can add it to your proyect in differents ways:

###### CSS
```css title="yourmain.css"
@import 'not-idea-ui/css/not-idea-ui.css';

/* or */

@import 'not-idea-ui/css/not-idea-ui.min.css';
```

###### TailwindCSS in order to use your own/customs styles (CSS)
```css title="yourmain.css"
@import 'not-idea-ui/tailwind/not-idea-ui.css';
```

###### TailwindCSS in order to use your own/customs styles (SCSS)
```css title="yourmain.scss"
@import 'not-idea-ui/main.scss';
```

<div className="alert is-warning">
  <h4 className="mb-4 font-semibold">Important note</h4>
  <p>When using the TailwindCSS preference, you need to add the following styles that I have predefined in your tailwind.config.js, to avoid errors that a certain class does not exist.</p>
</div>


###### Points to consider:

- Have at least the version "^3.3.x"
- In case you want to use it with SCSS. You need to install sass dependency in case you use ViteJS. If you use Webpack you need to install sass and sass-loader dependencies.
- If you want to use default styles and colors, follow this steps: [postcss.config.js](#custom-tailwindconfigjs)
- If you want to use your own styles and colors, follow this steps: [tailwind.config.js](#default-tailwindconfigjs)

## Default tailwind.config.js
### With ViteJS

To use the default config from the project you have to create a `postcss.config.cjs`. If you have already this file, just change the file extension from `.js` to `.cjs` in order the avoid the next error.

<span class="inline-block mb-4 text-sm text-red-500"><code>TypeError: Cannot read properties of undefined (reading 'config')</code>.</span>

This works for VanillaJS, ReactJS, and VueJS.

Add this line <span class="text-sky-500">`const defaultConfig = require("not-idea-ui/tailwind.config.cjs");`</span> to your `postcss.config.cjs`.

Should be seen:

```js title="postcss.config.cjs" theme="dark"
const defaultConfig = require("not-idea-ui/tailwind.config.cjs");

module.exports = {
  plugins: {
    tailwindcss: {
      config: defaultConfig
    },
    // ...
  },
}
```

And that is all, now it should be working without problem. :star:

### With Webpack

To use the default config from the project you have to create a `postcss.config.js` or `postcss.config.cjs`. It works with both option.

If you are getting this error: <span class="inline-block mb-4 text-sm text-red-500"><code>Error: Cannot find module 'tailwind.config.cjs'</code></span> or similar, just copy or download the config Tailwind file, it should work without problem.

This works for VueJS using Vue/Cli.

Add this line <span class="text-sky-500">`const defaultConfig = require("not-idea-ui/tailwind.config.cjs");`</span> to your `postcss.config.cjs`.

```js title="postcss.config.cjs" theme="dark"
const defaultConfig = require("not-idea-ui/tailwind.config.cjs");

module.exports = {
  plugins: {
    tailwindcss: {
      config: defaultConfig
    },
    // ...
  },
}
```

If you are using `create-react-app` for the project in ReactJS, you have to put the `tailwind.config.js` file in the root of the project, and remove the defaultConfig from `postcss.config.js`

```js title="postcss.config.js" theme="dark"
module.exports = {
  plugins: {
    tailwindcss: {},
    // ...
  },
}
```

And that is all, now it should be working without problem. :star:

## Custom tailwind.config.js

To use your custom `tailwind.config.js` you have to remove the default config from your `postcss.config.js` and leave it blank, as you can see below:

```js title="postcss.config.js" theme="dark"
module.exports = {
  plugins: {
    tailwindcss: {},
    // ...
  },
}
```

After that, run `tailwindcss init` to create blank `tailwind.config.js`. Now you have to fill the tailwind config with your `primary, secondary, and tertiary` colors and some custom classes to avoid some errors.

See the code and example below, be sure to copy every class and color. (<a href="https://github.com/IsraelDCastro/not-idea-ui/blob/master/tailwind.config.cjs" class="link info">For any reference you can see the default config</a>)

```js title="tailwind.config.cjs" theme="dark"
module.exports = {
  // ...
  theme: {
    extend: {
      colors: {
        primary: {
          50: "...",
          100: "...",
          200: "...",
          300: "...",
          400: "...",
          500: "...",
          600: "...",
          700: "...",
          800: "...",
          900: "...",
          950: "..."
        },
        secondary: {
          50: "...",
          100: "...",
          200: "...",
          300: "...",
          400: "...",
          500: "...",
          600: "...",
          700: "...",
          800: "...",
          900: "...",
          950: "..."
        },
        tertiary: {
          50: "...",
          100: "...",
          200: "...",
          300: "...",
          400: "...",
          500: "...",
          600: "...",
          700: "...",
          800: "...",
          900: "...",
          950: "..."
        }
      },
      transitionDuration: {
        0: "0ms",
        400: "400ms",
        600: "600ms"
      },
      zIndex: {
        "-1": -1,
        1: 1,
        5: 5
      },
      inset: {
        "-100": "-100%"
      },
      screens: {
        tablet: "989px",
        "max-md": { max: "988px" },
        mb: { max: "500px" }
      },
      borderWidth: {
        6: "6px"
      },
      boxShadow: {
        alert: "0 6px 26px rgba(0, 0, 0, 0.1)",
        base: "0 3px 20px rgba(0, 0, 0, 0.05)"
      },
      minHeight: {
        24: "6rem",
        "3-5": "14px"
      },
      listStyleType: {
        circle: "circle"
      }
    }
  },
  // ...
}
```
And that is all, now it should be working without problem. :star:


CHANGELOG.md
# Not Idea UI Changelog

---
### Version 2.2.0

- Add default font-familty to tailwind.config.cjs
- List: It were added more designs. Also, a todo list design. [View doc](https://not-idea-ui.netlify.app/docs/components/lists)
- Badges: It were added sizes for badges and more colors. [View doc](https://not-idea-ui.netlify.app/docs/components/badges)
- Tooltips: You can easily add tooltips to your buttons, cards, or any other element using CSS. Here's how to do it:
 [View doc](#!)

### Version 2.1.20

- Tailwind CSS update to the last version, secondary color was changed to `Persian Green`. According to the recommendation of Tailwind CSS, 950 colors were added to primary, secondary, and tertiary.
- Fixed submenu in sidebar for .rounded-links
- Form with datafield
- Removed sidebar with for more customization by user. More colors design added
- .double-field changed to grid css
- Cards: New design of cards here added. [Docs](https://not-idea-ui.netlify.app/docs/components/cards/default-cards#card-with-image-content)
- Buttons: Link primary, secondary, tertiary, white, dark, grey buttons; all color links added. [Links](https://not-idea-ui.netlify.app/docs/components/buttons/button-utilities#links)

### Version 2.1.3

- The image size in the product card was fixed. It was set up a default size for every image.

### Version 2.1.2

- Buttons icons were fixed and add to the [documentation](https://not-idea-ui.netlify.app/docs/components/accordions).

### Version 2.1.1

Some errors fixed on small screens in these components: **Galleries, Dropdown, Modal, Tabs, and Accordions**.

### Version 2.1.0

#### New components

- **Dropdown**: Display dropdowns as some easy ways to add more options and links to your project. Good dropdowns for displaying lists of links and more in a few easy ways by hover or focus. [See component](https://not-idea-ui.netlify.app/docs/components/dropdown)
- **Galleries**: Gallery components for your website, especially for the sections where you want to put a group of images. [See component](https://not-idea-ui.netlify.app/docs/components/galleries)

#### Updated components

- **Accordions**: Had been added more designs to this section. [See component](https://not-idea-ui.netlify.app/docs/components/accordions)
- **Alerts**: Had been added more designs to this section. Such as alerts with icons, alerts with a left border, and right button options. [See component](https://not-idea-ui.netlify.app/docs/components/alerts)
- **Badges**: Had been added more designs to this section and changed classes. The class for rounded corners was changed from `rounded` to `rounded-xl` as default. classes as primary, secondary, tertiary and others were added `is-` as the prefix. Classes were redesigned for much better simplicity. Class is-info color was changed from blue to sky. [See component](https://not-idea-ui.netlify.app/docs/components/badges/)
- **Hover cards**: Had been added a new design. [See component](https://not-idea-ui.netlify.app/docs/components/cards/hover-cards#card-image-hover)
- **Product cards**: Had been added a new design. [See component](https://not-idea-ui.netlify.app/docs/components/cards/product-cards#product-card-hover-buttons)
- **Form / Inputs**: Had been added a new styles and sizes. [See component](https://not-idea-ui.netlify.app/docs/layouts/form#input-styles)
- **Modals / Dialogs**: Had been added a new designs. [See component](https://not-idea-ui.netlify.app/docs/components/modals#right-modal)

### Version 2.0.0

#### New components

> - Progress
> - Breadcrumb
> - Modal / Dialogs

#### New improvements

> - Navbar has the option for sticky menu.
> - Buttons has link button design.
> - The doc was redesigned for a better experience and more stuff.

#### Bugfix

> - The menu on mobile had a transparent background, and now has a white background.


tailwind.config.js
module.exports = {
  theme: {
    extend: {
      colors: {
        "c-base": "#222",
        primary: {
          50: "#CEE9FF",
          100: "#B7D8F7",
          200: "#A0C7EF",
          300: "#8AB5E7",
          400: "#73A4DF",
          500: "#5C93D8",
          600: "#4582D0",
          700: "#2F70C8",
          800: "#185FC0",
          900: "#014EB8"
        },
        secondary: {
          50: "#FFF9F5",
          100: "#FFE9DA",
          200: "#FFD8BF",
          300: "#FFC8A3",
          400: "#FFB888",
          500: "#FFA76D",
          600: "#FF9752",
          700: "#FF8736",
          800: "#FF761B",
          900: "#FF6600"
        },
        tertiary: {
          50: "#FAF6FE",
          100: "#E6DBEF",
          200: "#D3BFE0",
          300: "#BFA4D1",
          400: "#AB89C2",
          500: "#986DB2",
          600: "#8452A3",
          700: "#703794",
          800: "#5D1B85",
          900: "#490076"
        }
      },
      transitionDuration: {
        0: "0ms",
        400: "400ms",
        600: "600ms"
      },
      zIndex: {
        "-1": -1,
        1: 1,
        5: 5
      },
      inset: {
        "-100": "-100%"
      },
      screens: {
        tablet: "989px",
        "max-md": { max: "988px" },
        mb: { max: "500px" }
      },
      borderWidth: {
        6: "6px"
      },
      boxShadow: {
        alert: "0 6px 26px rgba(0, 0, 0, 0.1)",
        base: "0 3px 20px rgba(0, 0, 0, 0.05)"
      },
      minHeight: {
        24: "6rem",
        "3-5": "14px"
      },
      listStyleType: {
        circle: "circle"
      },
      backgroundImage: (theme) => ({
        "card-image": "url('/images/galaxy.webp')"
      })
    },
  },
  variants: {
    extend: {},
  },
  plugins: [],
}
